ddtrace/tracer/textmap.go

```diff
@@ -264,46 +265,68 @@ func (p *chainedPropagator) Inject(spanCtx ddtrace.SpanContext, carrier interfac
-			// A local trace context has already been extracted.
-			pw3c, isW3C := v.(*propagatorW3c)
-			if !isW3C {
-				continue // Ignore other propagators.
-			}
-			w3cCtx, err := pw3c.Extract(carrier)
-			if err == nil && w3cCtx.(*spanContext).TraceID128() == ctx.(*spanContext).TraceID128() {
-				pw3c.propagateTracestate(ctx.(*spanContext), w3cCtx.(*spanContext))
-				if w3cCtx.SpanID() != ctx.SpanID() {
-					var ddCtx *spanContext
-					if ddp := getDatadogPropagator(p); ddp != nil {
-						if ddSpanCtx, err := ddp.Extract(carrier); err == nil {
-							ddCtx, _ = ddSpanCtx.(*spanContext)
+		extractedCtx, err := v.Extract(carrier)
+		if ctx != nil { // A local trace context has already been extracted
+			if err == nil {
+				propagatorType, unkPropagatorType := getPropagatorType(v)
+				if unkPropagatorType != nil { // If the propagator type is not one that we expect, ignore this extraction
+					continue
+				}
+				if extractedCtx.(*spanContext).TraceID128() == ctx.(*spanContext).TraceID128() {
+					if propagatorType == "tracecontext" {
+						v.(*propagatorW3c).propagateTracestate(ctx.(*spanContext), extractedCtx.(*spanContext)) //Sending tracestate to the ctx to be returned
+						if extractedCtx.SpanID() != ctx.SpanID() {
+							var ddCtx *spanContext
+							if ddp := getDatadogPropagator(p); ddp != nil {
+								if ddSpanCtx, err := ddp.Extract(carrier); err == nil {
+									ddCtx, _ = ddSpanCtx.(*spanContext)
+								}
+							}
+							overrideDatadogParentID(ctx.(*spanContext), extractedCtx.(*spanContext), ddCtx)
+						}
+					}
+				} else {
+					var flags uint32
+					if tracer := extractedCtx.(*spanContext).trace; tracer != nil {
+						if flags = uint32(*tracer.priority); flags > 0 { // Set the flags based on the sampling priority
+							flags = 1
+						} else {
+							flags = 0
 						}
 					}
-					overrideDatadogParentID(ctx.(*spanContext), w3cCtx.(*spanContext), ddCtx)
+					link := ddtrace.SpanLink{TraceID: extractedCtx.(*spanContext).TraceID(), TraceIDHigh: extractedCtx.(*spanContext).TraceIDUpper(), SpanID: extractedCtx.(*spanContext).SpanID(), Attributes: map[string]string{"reason": "terminated_context", "context_headers": propagatorType}, Flags: flags}
+					if propagatorType == "tracecontext" {
+						link.Tracestate = extractedCtx.(*spanContext).trace.propagatingTag(tracestateHeader)
+					}
+					links = append(links, link)
 				}
 			}
-			break
-		}
-		var err error
-		ctx, err = v.Extract(carrier)
-		if ctx != nil {
-			if p.onlyExtractFirst {
-				// Return early if the customer configured that only the first successful
-				// extraction should occur.
-				return ctx, nil
+		} else { // This is the first extracted trace context
+			ctx = extractedCtx
+			if ctx != nil {
+				if p.onlyExtractFirst {
+					// Return early if the customer configured that only the first successful
+					// extraction should occur.
+					return ctx, nil
+				}
+			} else if err != ErrSpanContextNotFound {
```

Comment:
I don't remember what `err` this is handling. Is it possible to move the error check after the `err` is assigned?